backend/mora/auth/sso.py

```diff
@@ -5,38 +5,273 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
-
+import base64
+import contextlib
+import functools
 import os
+from urllib import parse
 
 import flask
-import flask_saml
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+from onelogin.saml2.constants import OneLogin_Saml2_Constants
+from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser
+from onelogin.saml2.response import OneLogin_Saml2_Response
+from onelogin.saml2.xml_utils import OneLogin_Saml2_XML
 
 from . import base
-from . import tokens
-from .. import settings
 
 basedir = os.path.dirname(__file__)
 
+blueprint = flask.Blueprint('sso', __name__, static_url_path='',
+                            url_prefix='/saml')
+
+
+class AuthException(Exception):
```

Comment:
AFAICT you only instantiate this to raise it — shouldn't we use it, or check it, or something?